### PR TITLE
Make DOG work with Miniflare v2

### LINKED ISSUE #14: Usage with miniflare v2
Miniflare is throwing error when creating `new Request` using relative url. Thus I'm getting this error `Failed to parse URL from /~$~/identify` when running this.
```js
 const room = await identify(gid, reqid, {
    parent: env.LOBBY,
    child: env.ROOM,
  })
```

### PR DESCRIPTION
**Pull request by nikolalsvk**
Hey, thanks for the awesome library. I am looking into using it but, unfortunately, it doesn't work well with [Miniflare](https://github.com/cloudflare/miniflare) v2 (I'm on version 2.3.0 to be more specific). Here's the issue I'm trying to fix with this PR https://github.com/cloudflare/dog/issues/14

### How to test:
- Get a project with Miniflare v2 (I can create some, just let me know)
- Run `npm install` in the root of this project, and also inside `bin` dir with `cd bin && npm install`
- Run `npm run build`
- Then, in your project, add the local npm package via npm like `npm install ../path/to/dog` or `yarn add file:../../path/to/dog`
- Try running your Durable Object Group, it should work

Hope it helps, let me know what you think 🤞 

---

Closes #14

### CODE REVIEW

File: src/group.ts [hunk 1]
```diff
@@ -55,7 +55,7 @@ export abstract class Group<T extends ModuleWorker.Bindings> implements DOG.Grou
 	 */
 	async fetch(input: RequestInfo, init?: RequestInit): Promise<Response> {
 		let request = new Request(input, init);
-		let { pathname } = new URL(request.url, 'foo://');
+		let { pathname } = new URL(request.url);
```
Comment (lukeed) [inline src/group.ts:58]: This is the only line that had to change. Can you please revert all other changes and make this:

```ts
let { pathname } = new URL(request.url, 'http://dog');
```

That should do the trick

**Review by lukeed — CHANGES_REQUESTED** [review]

File: src/group.ts (hunk 1 above)
Comment (nikolalsvk) [reply src/group.ts:58]: I tried, but it is failing with only the change you suggested. The real DOG 'breaks' in the `src/glue.ts` when it tries to make a new request with a non-valid URL here https://github.com/cloudflare/dog/blob/30517cd97bc0911dce8b819296a5cee59ea01ab7/src/glue.ts#L9

Comment (lukeed) [reply src/group.ts:58]: Ah right, then can update that line too.

```ts
let request = new Request(ROUTES.IDENTIFY, 'http://dog');
```

Comment (nikolalsvk) [reply src/group.ts]: I ended up doing:

```ts
let request = new Request(new URL(ROUTES.IDENTIFY, 'http://dog'));
```

WDYT?

Comment (lukeed) [reply src/group.ts]: oh brain fart 😅 I thought I was typing `new URL` – that's perfect, thanks!

File: src/glue.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@ import type * as DOG from 'dog';
 export const identify: typeof DOG.identify = async function (gid, rid, family) {
 	let group = family.parent.get(gid);
 
-	let request = new Request(ROUTES.IDENTIFY);
+  let request = new Request(new URL(ROUTES.IDENTIFY, 'http://dog'));
```
Comment (lukeed) [inline src/glue.ts:9]:
```suggestion
	let request = new Request('http://dog' + ROUTES.IDENTIFY);
```

File: src/group.ts [hunk 3]
```diff
@@ -55,7 +55,7 @@ export abstract class Group<T extends ModuleWorker.Bindings> implements DOG.Grou
 	 */
 	async fetch(input: RequestInfo, init?: RequestInit): Promise<Response> {
 		let request = new Request(input, init);
-		let { pathname } = new URL(request.url, 'foo://');
+    let { pathname } = new URL(request.url, 'http://dog');
```
Comment (lukeed) [inline src/group.ts:58]:
```suggestion
    	let { pathname } = new URL(request.url, 'http://dog');
```

**Review by lukeed — APPROVED** [review]

**Pull request merged**